backend/proxy/secure_socks_proxy.go

```diff
@@ -83,7 +83,7 @@ type cfgProxyWrapper struct {
 // and the datasource options specify to use the proxy
 func (p *cfgProxyWrapper) SecureSocksProxyEnabled() bool {
 	// it cannot be enabled if it's not enabled on Grafana
-	if p.opts == nil {
+	if p.opts == nil || p.opts.ClientCfg == nil {
```

Comment:
I think it could be helpful to add to the comment above to cover the scenario you mentioned in the PR description

Reply:
Done, let me know if the comment is clear enough.

Reply:
Actually (apologies), I think maybe it's better to do instead:
```suggestion
	if p.opts == nil || !p.opts.Enabled {
```

That should have the same effect. We can skip the change to the comment then too.

Reply:
Ahhh I think that's somewhat of a bug then in that case. `Enabled` ideally should only be set when both the datasource AND Grafana has PDC enabled. Maybe then let's change to the following:
```suggestion
// the secure proxy can only be used if it's enabled on both the datasource connection and the client (Grafana server)
	if p.opts == nil || !p.opts.Enabled || p.opts.ClientCfg == nil { 
```